crates/ruff_server/src/session/settings.rs

```diff
@@ -38,6 +38,22 @@ pub(crate) struct ResolvedEditorSettings {
     pub(super) ignore: Option<Vec<RuleSelector>>,
     pub(super) exclude: Option<Vec<String>>,
     pub(super) line_length: Option<LineLength>,
+    pub(super) configuration_resolution_strategy: ConfigResolutionStrategy,
+}
+
+/// Determines how multiple conflicting configurations should be resolved - in this
+/// case, the configuration from the client settings and configuration from local
+/// `.toml` files (aka 'workspace' configuration).
+#[derive(Clone, Copy, Debug, Deserialize)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[serde(rename_all = "camelCase")]
+pub(crate) enum ConfigResolutionStrategy {
+    /// The default strategy - configuration set in the editor takes priority over
+    /// workspace configuration set in `.toml` files.
+    Default,
```

Comment:
I think we should have a name here that describes the strategy, rather than using `Default`.

Reply:
Could these be "Editor" and "Workspace" (or "Filesystem"), if the setting were called "configurationPreference"?

Reply:
(I'd also suggest making it `#[default]`.)

Reply:
I like Charlie's suggestion. My first thought was `PrioritizeEditor`. I think I would not call it resolution, because I associate resolution with finding the configuration. Althought that's probably just me.

Reply:
Yeah I think I'd be fine approving either `ConfigurationPreference` or `ConfigurationPrecedence`, with options like:

- `Editor`
- `Workspace`

Alternatively, if we _do_ want an option that uses the editor and _ignores_ workspace configuration (which is what I initially expected it to do), we could have:

- `EditorFirst` (equivalent to the `Default` here)
- `FilesystemFirst` (equivalent to `PrioritizeWorkspace` here)
- `EditorOnly` (use editor settings, ignore on-disk configuration)

Reply:
I like `ConfigurationPreference` and `EditorFirst`/`FilesystemFirst`/`EditorOnly` 👍